src/components/organisms/UiMenu/UiMenu.vue

```diff
@@ -62,139 +53,175 @@ export interface MenuProps {
+  } else {
+    usedMenuTemplateRefs.value.forEach((item) => {
+      // eslint-disable-next-line no-param-reassign
+      item.tabindex = 0;
+    });
+  }
+});
+onMounted(async () => {
+  await nextTick();
+  if (props.enableKeyboardNavigation
+      && usedMenuTemplateRefs.value.length > 0) {
+    setNegativeTabindexForNonInitialMenuItems();
+  }
+  if (process.env.NODE_ENV !== 'production') {
+    if (props.enableKeyboardNavigation
+        && usedMenuTemplateRefs.value.length < 1
+    ) {
+      console.warn('@infermedica/component-library: use itemsTemplateRefs to pass UiMenuItems template refs.');
+    }
+  }
+});
+const lastFocusedMenuItemTemplateRefs = ref<ElementRef | null>(null);
+const handleMenuItemFocus = async (element: ElementRef | InstanceType<typeof UiMenuItem> | undefined) => {
+  if (element && initialElement.value) {
+    await focusElement(element?.itemTemplateRefs?.content?.$el, true);
+    initialElement.value.tabindex = -1;
+  }
+};
+const handleMenuKeyDown = async ({ key }: KeyboardEvent) => {
+  if (!props.enableKeyboardNavigation) {
+    return;
+  }
+  const activeElement = focusedElement.value;
   switch (key) {
     case 'ArrowUp':
-      if (prevMenuItem.value && focusedMenuItem.value) {
-        focusedMenuItem.value.tabindex = -1;
-        prevMenuItem.value.tabindex = 0;
-        focusElement(prevMenuItem.value.$el.querySelector('.ui-button'));
-      }
+      handleMenuItemFocus(prevElement.value);
       break;
     case 'ArrowDown':
-      if (nextMenuItem.value && focusedMenuItem.value) {
-        focusedMenuItem.value.tabindex = -1;
-        nextMenuItem.value.tabindex = 0;
-        focusElement(nextMenuItem.value.$el.querySelector('.ui-button'));
-      }
+      handleMenuItemFocus(nextElement.value);
       break;
     case 'Home':
     case 'PageUp':
-      if (firstMenuItem.value && focusedMenuItem.value) {
-        focusedMenuItem.value.tabindex = -1;
-        firstMenuItem.value.tabindex = 0;
-        focusElement(firstMenuItem.value.$el.querySelector('.ui-button'));
-      }
+      handleMenuItemFocus(usedMenuTemplateRefs.value.at(0));
       break;
     case 'End':
     case 'PageDown':
-      if (lastMenuItem.value && focusedMenuItem.value) {
-        focusedMenuItem.value.tabindex = -1;
-        lastMenuItem.value.tabindex = 0;
-        focusElement(lastMenuItem.value.$el.querySelector('.ui-button'));
-      }
+      handleMenuItemFocus(usedMenuTemplateRefs.value.at(-1));
       break;
     case 'Tab':
-      const lastFocusedElement = ref(focusedMenuItem.value);
-      setTimeout(() => {
-        if(!lastFocusedElement.value) return;
-        lastFocusedElement.value.tabindex = -1;
-        if (selectedMenuItem.value) {
-          selectedMenuItem.value.tabindex = 0;
-          return;
-        }
-        if (firstMenuItem.value) {
-          firstMenuItem.value.tabindex = 0;
-        }
-      }, 0);
+      if (initialElement.value === activeElement) {
+        return;
+      }
```

Comment:
Why not like that? Doesn't eslint allow that?

```suggestion
      if (initialElement.value === activeElement) return;
```

Reply:
It's a personal preference, a few years ago I read some books where they recomended using always `{}` brackets and I use it 🙈